modules/oe_content_person/oe_content_person.module

```diff
@@ -42,3 +46,102 @@ function oe_content_person_node_presave(EntityInterface $entity) {
+  $user_input = $form_state->getUserInput();
+  if (isset($user_input['oe_person_type'])) {
+    $type = $user_input['oe_person_type'];
+  }
+  if ($type !== 'eu') {
+    $entity_form['oe_role_reference']['#attributes']['style'] = 'display:none';
+    $entity_form['oe_acting']['#attributes']['style'] = 'display:none';
+    $entity_form['oe_role_name']['widget'][0]['value']['#required'] = TRUE;
+  }
+
+  if ($type !== 'non_eu') {
+    $entity_form['oe_role_name']['#attributes']['style'] = 'display:none';
+    $entity_form['oe_role_reference']['widget'][0]['target_id']['#required'] = TRUE;
+  }
+
+  // Set states based on "What type of person are you adding?" field.
+  ContentFormUtilities::toggleFieldByValue($entity_form, 'oe_person_type', 'oe_role_name', 'non_eu');
+  ContentFormUtilities::toggleFieldByValue($entity_form, 'oe_person_type', 'oe_role_reference', 'eu');
+  ContentFormUtilities::toggleFieldByValue($entity_form, 'oe_person_type', 'oe_acting', 'eu');
+
+  $entity_form['oe_role_name']['widget'][0]['value']['#states']['required'] = [
+    ':input[name="oe_person_type"]' => [
+      'value' => 'non_eu',
+    ],
+  ];
+  $entity_form['oe_role_reference']['widget'][0]['target_id']['#states']['required'] = [
+    ':input[name="oe_person_type"]' => [
+      'value' => 'eu',
+    ],
+  ];
+}
+
+/**
+ * Validation callback for a "oe_person_jobs" element.
+ *
+ * @param array $element
+ *   The element being processed.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ * @param array $complete_form
+ *   The complete form structure.
+ */
+function oe_content_person_oe_person_jobs_validate(array &$element, FormStateInterface $form_state, array &$complete_form): void {
+  $type = $form_state->getValue(['oe_person_type', 0, 'value']);
+
+  $field_values = $form_state->getValue(['oe_person_jobs', 'form', 'inline_entity_form', 'entities']);
+  if (!empty($field_values)) {
+    // Clear hidden fields.
+    $delta = key($field_values);
+    if ($type !== 'non_eu') {
+      $form_state->setValue(['oe_person_jobs', 'form', 'inline_entity_form', 'entities', $delta, 'form', 'oe_role_name'], []);
+    }
+    if ($type !== 'eu') {
+      $form_state->setValue(['oe_person_jobs', 'form', 'inline_entity_form', 'entities', $delta, 'form', 'oe_role_reference'], []);
+      $form_state->setValue(['oe_person_jobs', 'form', 'inline_entity_form', 'entities', $delta, 'form', 'oe_acting'], []);
+    }
+  }
+
+  // Enforce to update Role field on the existing jobs if person type has been
+  // changed.
+  foreach (Element::children($complete_form['oe_person_jobs']['widget']['entities']) as $key) {
+    $item = $complete_form['oe_person_jobs']['widget']['entities'][$key];
+    if (!empty($item['#entity']) && !_oe_content_person_validate_person_job_entity($item['#entity'], $type)) {
+      $form_state->setError($element, t('Role field is required in the job @label', ['@label' => $item['#entity']->label()]));
```

Comment:
We could maybe be more descriptive, so we explain the nature of the problem to the editor, something like:

```
The role "@label" is not compatible with the type of person currently selected. Please edit the related job entry and fix its role accordingly. 
```

Reply:
Message has been updated.